log4j-slf4j20-impl/src/main/java/org/apache/logging/slf4j/Log4jMDCAdapter.java

```diff
@@ -16,16 +16,87 @@
  */
 package org.apache.logging.slf4j;
 
+import java.util.ArrayDeque;
+import java.util.Deque;
 import java.util.Map;
+import java.util.Objects;
 
+import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.util.Strings;
 import org.slf4j.spi.MDCAdapter;
 
 /**
  *
  */
 public class Log4jMDCAdapter implements MDCAdapter {
 
+    private static final Logger LOGGER = StatusLogger.getLogger();
+
+    private static final char QUOTE_CHAR = '\"';
+    private static final char ESCAPE_CHAR = '\\';
+    private static final char ARRAY_START = '[';
+    private static final char ARRAY_SEP = ',';
+    private static final char ARRAY_END = ']';
+    private static final String EMPTY_ARRAY = "[]";
+
+    /**
+     * Quotes a string using a simplified JSON quoting mechanism.
+     */
+    static void quoteString(final CharSequence value, final StringBuilder output) {
```

Comment:
In accordance with my comment below, I would rather move `JsonWriter` from JTL to core and reuse it here.